### PR TITLE
Update GCP exporters to v0.48.2

### PR DESCRIPTION
**Pull request by damemi**
**Description:**
This updates the GCP dependencies to v0.48.2. Full release notes for this release are available at https://github.com/GoogleCloudPlatform/opentelemetry-operations-go/releases/tag/v0.48.2

**Link to tracking Issue:** <Issue number if applicable>

**Testing:** Downstream unit/integration tests

**Documentation:** n/a

Fixes https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/35206

### CODE REVIEW

**Comment by dashpole** [comment]
looks like the GMP exporter needs to be updated.  Also remember to mark the exporters mutating.  That should be using `exporterhelper.WithCapabilities(consumer.Capabilities{MutatesData: true})`

**Review by songy23 — COMMENTED** [review]
> go.mod/go.sum deps changes detected, please run "make gotidy" and commit the changes in this PR.

File: .chloggen/gcp-0482.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+# Use this changelog template to create an entry for release notes.
+
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: enhancement
+
+# The name of the component, or a single word describing the area of concern, (e.g. filelogreceiver)
+component: googlecloudexporter
+
+# A brief description of the change.  Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: Google Cloud and GMP exporters are marked as mutating.
```
Comment (songy23) [inline .chloggen/gcp-0482.yaml:10]: nit: what we usually do is to have 2 changelog entries, one for googlecloudexporter and one for googlemanagedprometheusexporter

**Review by songy23 — APPROVED** [review]

**Review by dashpole — APPROVED** [review]

**Review by codeboten — APPROVED** [review]
Needs a rebase and it should be good to merge after

**Comment by dashpole** [comment]
@damemi please rebase when you get the chance.

**Comment by damemi** [comment]
Rebased, thanks!

**Pull request merged**